### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by PallHaraldsson**
[skip ci]

### CODE REVIEW

**Comment by PallHaraldsson** [comment]
I missed at least one more: "insted". Make sure my "most weight" change is appropriate.

Some questions, for e.g. "Instead of push!() and setindex!()" I can see why you rather want e.g. push. It's not wrong (to be explicit in meaning, I guess why you did this), but I'm not sure push! is either? I was thinking are these data-structures a drop-in replacement for non-persistent ones. Then for lots of code you might already have you could have e.g. push!, and while you want to change to push (I guess it's faster), it might be nice not to have to. Also if for some reason you want or need to go back to the old data structures, then it would be easier. Maybe either change the API to only have push! or add it as an alias?

"pfill(), pones(), and pzeros(), which are all similar to their non-persistent counterparts fill(), ones(), and zeros()"

You must do that to get the correct type back, but again I guess you could want e.g. fill! working.

Minor issues:
"the log function"->"the logmsg function" or "that log"?

"Actors are only sent # functions"

"sent to functions"? I'm guessing, not sure I understand Actors, if this wording intentional, or could be even completely different.


Seems hyphens intended, maybe not needed:
"managing side-effects"

"Arbitrarily-deep nested persistent [..] data-structure"

the latter needs not be there, but for deep, I'm conflicted, should it be before or after or both...

**Comment by noahbenson** [comment]
Thanks for the typo fixes—the keyboard on the MacBook Air that I use for 95% of my editing is dying, so this code base is full of doubled and missing letters (not to mention my own typos).

I think you have a misconception about what the data structures in this library do; for example, I disagree that `push!` is a synonym for `push`. The function `push` doesn't modify the underlying data structure while `push!` does. You cannot use the data structures in `Air` as drop-in replacements for mutable data structures. For example, this code:
```julia
dict = Dict(:a => 1, :b => 2)
push!(dict, :c => 3)
:c in dict
```
will yield `true`, but this code:
```julia
dict = PDict(:a => 1, :b => 2)
push(dict, :c => 3)
:c in dict
```
will yield false. To get it to yield `true`, you would need to change the second line to `dict = push(dict, :c => 3)`.

Similarly, the function `fill!` would be useless for persistent data structures because in order to use `fill!` you must write into the data structure. This would break the guarantee of persistence.

File: README.md [hunk 1]
```diff
@@ -197,7 +197,8 @@ julia> notes[]
 
 # We can define a function that sends a message to the log. The send function
 # requires that the first argument be a function, to be run in the actor's
-# thread, whose one parameter is the current value of actor and whose return
```
Comment (noahbenson) [inline README.md]: I think change this was a mistake?

File: README.md [hunk 2]
```diff
@@ -242,23 +243,23 @@ julia> notes[]
...
 
-* Completed plans
+* Completed plans:
   * Persistent data structures:
     * `PArray`, a persistent array type that mimics Julia's native `Array`
     * `PDict`, a persistent dictionary type that mimics Julia's native `Dict`
     * `PSet`, a persistent set type.
     * `PWSet`, a weighted persistent set type. Each element of a `PWSet` has a
-      weight, and the sete itself acts both as a piority queue (`first(pwset)`
-      always yields the element with the highest weight in `O(1)` time) and as
```
Comment (noahbenson) [inline README.md:257]: I disagree that "most weight" is correct in this context. The "weight" is a single numeric variable, not a physical corollary of something's mass.

File: README.md [hunk 3]
```diff
@@ -271,14 +272,14 @@ expert on testing multi-threaded code, some caution is advisable.
...
     * Thread-safe `Promise` types.
-* Plans that are not yet implemented
+* Plans that are not yet implemented:
   * **Improved Persistent Array Methods.** Currently, most methods of persistent
     arrays yield mutable Julia arrays instead of persistent arrays. This is
     largely because the `PArray` class does not explicitly overload these
     functions or the broadcasting functions in order to ensure that it creates
     `PArray` objects. For example, `PArray([1,2,3]) .+ 1` should yield a
     `PArray`; currently it yields an `Array`.
-  * **Forms.** Arbitrarily-deep nested persistent dictionaries and vectors are
```
Comment (noahbenson) [inline README.md]: I disagree that this is correct. I am not saying they are arbitrarily-deeply nested, I am saying that they are both nested and arbitrarily-deep.

**Comment by noahbenson** [comment]
I added some comments—if you fix these, I'll merge. Thanks again!

**Comment by PallHaraldsson** [comment]
Unrelated... but I liked your video:
https://www.youtube.com/watch?v=YuisKtLshs0

I know the fact is exploited in VR headsets with eye-tracking (but didn't know of the asymmetry, maybe my professor should have known and mentioned), but it seems could also be done for graphics without VR. I.e. just generate graphics with half vertical resolution overall (even the middle, unclear how close to the middle of the retina the asymmetry still holds), and since for games mostly watching in the middle maybe even a quarter resolution in the absolute edges of the screen or at least high up (e.g. helpful for ray-tracing).

**Comment by noahbenson** [comment]
Oh ha, thanks!  My professional work is not super-related to this library, but I'm essentially trying to get Julia ready to support the neuroscience work I do so that some day I can migrate from Python! 🙃

**Comment by PallHaraldsson** [comment]
You should probably merge this. I wasn't clear on what more changes you seemed to want with "I added some comments—if you fix these, I'll merge." I had forgotten about this PR, for now, and maybe you too. I can make a new PR (or add a commit here?) after you merge, if needed.

**Comment by noahbenson** [comment]
I added comments to this very pull request. You can see them in this page. If you update your pull request to accommodate the comments here, I will merge. Until you answer/fix the comments, however, I will not merge because I do not agree with all of the changes you proposed.

**Comment by PallHaraldsson** [comment]
If you mean the comments are here: https://github.com/noahbenson/Air.jl/pull/1#issuecomment-953126528

then I think it's about what I wrote on this page, not on the actual content of the change in my commit of the PR. I at least fail to locate the problem you have (is it about a change/addition in the commit or a further change you want?). I think this may just be a miscommunication, or Github lost some comment you made? You can click on the "Files changed" tab and then the plus in front of where you want a change and be explicit (like I just did). I'm willing to change when I understand the objection.

**Comment by noahbenson** [comment]
Oh, hmm, you are right—they are showing up as pending and not as committed. I was sure I committed them, but apparently that was for some other pull request 🤷
Anyway, sorry about that. I'm committing them now. They are pretty small—I would have just fixed them myself, but I didn't want to steal the commit credit.

**Review by noahbenson — CHANGES_REQUESTED** [review]

**Comment by noahbenson** [comment]
Great, thanks! 😁

**Pull request merged**